x/budget/keeper/budget.go

```diff
@@ -52,6 +55,24 @@ func (k Keeper) CollectBudgets(ctx sdk.Context) error {
 		}
 		for i, budget := range budgetsBySource.Budgets {
 			k.AddTotalCollectedCoins(ctx, budget.Name, budgetsBySource.CollectionCoins[i])
+
+			defer func() {
+				for _, collectionCoins := range budgetsBySource.CollectionCoins {
+					for _, coin := range collectionCoins {
+						if coin.Amount.IsInt64() {
+							telemetry.SetGaugeWithLabels(
+								[]string{types.ModuleName},
+								float32(coin.Amount.Int64()),
+								[]metrics.Label{
+									telemetry.NewLabel("collection_address", budget.CollectionAddress),
+									telemetry.NewLabel("denom", coin.Denom),
+								},
+							)
+						}
+					}
+				}
+			}()
+
```

Comment:
```suggestion

			defer func() {
				for _, coin := range budgetsBySource.CollectionCoins[i] {
					if coin.Amount.IsInt64() {
						telemetry.SetGaugeWithLabels(
							[]string{types.ModuleName},
							float32(coin.Amount.Int64()),
							[]metrics.Label{
								telemetry.NewLabel("collection_address", budget.CollectionAddress),
								telemetry.NewLabel("denom", coin.Denom),
							},
						)
					}
				}
			}()
```

when if  iterate `for _, collectionCoins := range budgetsBySource.CollectionCoins {`  without `i`, It could be emitted duplicated IMO, what do you think?

Reply:
I didn't pay attention to the `i` from the above for loop. That is certainly better logic. Thanks!